src/consensus/aft/raft.h

```diff
@@ -2095,7 +2124,10 @@ namespace aft
       progress_tracker->add_nonce_reveal(
         {r.term, r.idx}, r.nonce, from, node_count(), is_primary());
 
-      update_commit();
+      if (!public_only)
```

Comment:
Why don't we update commit in public only mode? How can we commit governance changes then?

Reply:
I will add a comment to the code - What is going on here is during public only mode we do not want to move commitment forward because of nonces via the message channels only from evidence that is written on the ledger.

Reply:
As discussed: update_commit should be safe for a sparse history, and we don't need to/want to validate the pre-catastrophe ledger.